runtime/vm/jvminit.c

```diff
@@ -4102,6 +4102,26 @@ registerVMCmdLineMappings(J9JavaVM* vm)
 	if (registerCmdLineMapping(vm, MAPOPT_XXONOUTOFMEMORYERROR_EQUALS, VMOPT_XDUMP_TOOL_OUTOFMEMORYERROR_EXEC_EQUALS, EXACT_MAP_WITH_OPTIONS) == RC_FAILED) {
 		return RC_FAILED;
 	}
+	/* Map -XX:LargePageSizeInBytes=  to -Xlp:codecache:pagesize= */
+	if (registerCmdLineMapping(vm, MAPOPT_XXLARGEPAGESIZEINBYTES_EQUALS, VMOPT_XLP_CODECACHE_PAGESIZE_EQUALS, MAP_WITH_INCLUSIVE_OPTIONS) == RC_FAILED) {
+		return RC_FAILED;
+	}
+	/* Map -XX:+UseLargePages to -Xlp */
+	if (registerCmdLineMapping(vm, MAPOPT_XXUSELARGEPAGES, VMOPT_XLP, EXACT_MEMORY_MATCH) == RC_FAILED) {
+		return RC_FAILED;
+	}
+	/* Map -XX:ParallelCMSThreads=N to -Xconcurrentbackground */
+	if (registerCmdLineMapping(vm, MAPOPT_XXPARALLELCMSTHREADS_EQUALS, VMOPT_XCONCURRENTBACKGROUND, EXACT_MAP_WITH_OPTIONS) == RC_FAILED) {
+		return RC_FAILED;
+	}
+	/* Map -XX:ConcGCThreads=N  to -Xconcurrentbackground */
+	if (registerCmdLineMapping(vm, MAPOPT_XXCONCGCTHREADS_EQUALS, VMOPT_XCONCURRENTBACKGROUND, EXACT_MAP_WITH_OPTIONS) == RC_FAILED) {
+		return RC_FAILED;
+	}
+	/* Map -XX:ParallelGCThreads=N  to -XgcthreadsN */
+	if (registerCmdLineMapping(vm, MAPOPT_XXPARALLELGCTHREADS_EQUALS, VMOPT_XGCTHREADS, EXACT_MAP_WITH_OPTIONS) == RC_FAILED) {
+		return RC_FAILED;
+	}
```

Comment:
@dmitripivkine This is very different than the way the GC usually processes commandline arguments.  Would you rather handle the mapping directly in the GC?

Reply:
Obviously I would like to see this parsing inside GC code... except we have an intention to keep parsing of `-XX` options in one place in VM. I am not familiar with this options mapping code. Is it just rebuild command line by replacing one option to another (for each mapped)?

Reply:
Comment is stale. Closing.